frontend_tests/node_tests/unread.js

```diff
@@ -263,19 +261,18 @@ run_test('num_unread_for_topic', () => {
     let count = unread.num_unread_for_topic(stream_id, 'lunch');
     assert.equal(count, 0);
 
-    const message = {
-        type: 'stream',
-        stream_id: stream_id,
-        topic: 'LuncH',
-        unread: true,
-    };
-
     // Put messages into list in reverse order to try to confuse
     // our sort.
     const num_msgs = 500;
     let i;
     for (i = num_msgs; i > 0; i -= 1) {
-        message.id = i;
+        const message = add_message({
```

Comment:
Can you add a quick note in the commit message explaining all the `add_message` changes?  Is the idea just to have more realistic testing?

Reply:
Sure. `unread_topic_counter` calls `message_store.get` now, so we need it to know about our fake messages.

Reply:
Why does it need to call `message_store.get`?

Reply:
Oh, I see, it's because we went from FoldDict to Map.